client.ts

```diff
@@ -1,44 +1,67 @@
 import { dial } from "deno";
 import { Connection } from "./connection.ts";
+import { Pool } from "./pool.ts";
 import { Query, QueryConfig, QueryResult } from "./query.ts";
-import { IConnectionParams, ConnectionParams } from "./connection_params.ts";
+import { ConnectionParams, IConnectionParams } from "./connection_params.ts";
 
 export class Client {
-  connection: Connection;
-  connectionParams: ConnectionParams;
+  private _connection: Connection;
+  private _connectionParams: ConnectionParams;
+  release: () => void;
 
-  constructor(config?: IConnectionParams | string) {
-    this.connectionParams = new ConnectionParams(config);
+  constructor(
+    config?: IConnectionParams | string | Connection,
+    release?: () => void
+  ) {
+    if (config instanceof Connection) {
+      this._connection = config;
+      this.release = function() {
+        release();
+        delete this._connection;
+        delete this.release;
+      };
+    } else {
+      this._connectionParams = new ConnectionParams(config);
+    }
   }
 
-  async connect() {
-    const { host, port } = this.connectionParams;
+  private async newConnection(connectionParams: ConnectionParams) {
```

Comment:
Let's keep previous method, I want to add come lifecycle control anyway.

Reply:
The issue is that you want to _not_ have that lifecycle control exposed on a pooled client.